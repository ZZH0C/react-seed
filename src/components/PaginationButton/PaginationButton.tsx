import React from 'react';

// import classNames from 'classnames';

interface PaginationButtonProps
  extends React.ButtonHTMLAttributes<HTMLElement> {
  isDisabled: boolean;
  isRight: boolean;
  onClickFunction: any;
}

export const PaginationButton: React.FC<PaginationButtonProps> = ({
  onClickFunction,
  isDisabled,
  isRight,
  children,
}) => {
  let buttonText;
  if (isRight) {
    buttonText = (
      <>
        {children}
        <i className={'icon-right-open-big'} />
      </>
    );
  } else {
    buttonText = (
      <>
        <i className={'icon-left-open-big'} />
        {children}
      </>
    );
  }

  return (
    <button
      onClick={onClickFunction}
      disabled={isDisabled}
      className={'btn btn-default btn-prevNext'}
    >
      {buttonText}
    </button>
  );
};
