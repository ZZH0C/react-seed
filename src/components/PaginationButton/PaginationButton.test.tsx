import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { PaginationButton } from './PaginationButton';

describe('components/ButtonLink', () => {
  const mockProps = {
    onClickFunction: () => {
      return null;
    },
    children: 'SOME_TEXT',
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render right disabled and match snapshot', () => {
    expect(
      shallow(
        <PaginationButton
          isDisabled={true}
          isRight={true}
          onClickFunction={mockProps.onClickFunction()}
        >
          {mockProps.children}
        </PaginationButton>,
      ),
    ).toMatchSnapshot();
  });
  it('should render left disabled and match snapshot', () => {
    expect(
      shallow(
        <PaginationButton
          isDisabled={true}
          isRight={false}
          onClickFunction={mockProps.onClickFunction()}
        >
          {mockProps.children}
        </PaginationButton>,
      ),
    ).toMatchSnapshot();
  });
  it('should render right active and match snapshot', () => {
    expect(
      shallow(
        <PaginationButton
          isDisabled={false}
          isRight={true}
          onClickFunction={mockProps.onClickFunction()}
        >
          {mockProps.children}
        </PaginationButton>,
      ),
    ).toMatchSnapshot();
  });
  it('should render left active and match snapshot', () => {
    expect(
      shallow(
        <PaginationButton
          isDisabled={false}
          isRight={false}
          onClickFunction={mockProps.onClickFunction()}
        >
          {mockProps.children}
        </PaginationButton>,
      ),
    ).toMatchSnapshot();
  });
});
