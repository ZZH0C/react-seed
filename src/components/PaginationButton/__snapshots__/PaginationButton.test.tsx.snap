// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ButtonLink should render left active and match snapshot 1`] = `
<button
  className="btn btn-default btn-prevNext"
  disabled={false}
  onClick={null}
>
  <i
    className="icon-left-open-big"
  />
  SOME_TEXT
</button>
`;

exports[`components/ButtonLink should render left disabled and match snapshot 1`] = `
<button
  className="btn btn-default btn-prevNext"
  disabled={true}
  onClick={null}
>
  <i
    className="icon-left-open-big"
  />
  SOME_TEXT
</button>
`;

exports[`components/ButtonLink should render right active and match snapshot 1`] = `
<button
  className="btn btn-default btn-prevNext"
  disabled={false}
  onClick={null}
>
  SOME_TEXT
  <i
    className="icon-right-open-big"
  />
</button>
`;

exports[`components/ButtonLink should render right disabled and match snapshot 1`] = `
<button
  className="btn btn-default btn-prevNext"
  disabled={true}
  onClick={null}
>
  SOME_TEXT
  <i
    className="icon-right-open-big"
  />
</button>
`;
